From stable-bounces@linux.kernel.org Tue Feb 27 11:15:20 2007
From: David Stevens <dlstevens@us.ibm.com>
Date: Tue, 27 Feb 2007 11:14:00 -0800 (PST)
Subject: Fix anycast procfs device leak
To: stable@kernel.org
Cc: bunk@stusta.de
Message-ID: <20070227.111400.17864504.davem@davemloft.net>

From: David Stevens <dlstevens@us.ibm.com>

[IPV6]: /proc/net/anycast6 unbalanced inet6_dev refcnt

From: David Stevens <dlstevens@us.ibm.com>

Reading /proc/net/anycast6 when there is no anycast address
on an interface results in an ever-increasing inet6_dev reference
count, as well as a reference to the netdevice you can't get rid of.

From: David Stevens <dlstevens@us.ibm.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/ipv6/anycast.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.20.1.orig/net/ipv6/anycast.c
+++ linux-2.6.20.1/net/ipv6/anycast.c
@@ -462,6 +462,7 @@ static inline struct ifacaddr6 *ac6_get_
 			break;
 		}
 		read_unlock_bh(&idev->lock);
+		in6_dev_put(idev);
 	}
 	return im;
 }
